specification/quantumleap.yml

```diff
@@ -0,0 +1,500 @@
+      summary: "History of N attributes of a given entity instance."
+      description: "For example, query the average pressure, temperature and humidity of this month in the weather station WS1."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/attrs'
+        - $ref: '#/parameters/aggrMethod'
+        - $ref: '#/parameters/aggrPeriod'
+        - $ref: '#/parameters/options'
+        - $ref: '#/parameters/from'
+        - $ref: '#/parameters/to'
+        - $ref: '#/parameters/lastN'
+        - $ref: '#/parameters/limit'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+              attrName:
+                type: string
+              index:
+                type: array
+                items:
+                  type: number
+                description: "Array of the timestamps of the requested data. Values are UTC milliseconds since 1970-01-01T00:00:00Z."
+              values:
+                type: array
+                items:
+                  type: number
+                description: "Array of values of the selected attribute, in the same corresponding order of the 'index' array."
+          examples:
+            application/json:
+              {
+                "id": "Kitchen1",
+                "attrName": "temperature",
+                "index": [1510826981388, 1510826990438, 1510826999725],
+                "values": [24.1, 25.3, 26.7]
+              }
+
+  /entities/{entityId}/value:
+    get:
+      operationId: reporter.reporter.query_1T1ENA_value
+      summary: "History of N attributes (values only) of a given entity instance."
+      description: "For example, query the average pressure, temperature and humidity (values only, no metadata) of this month in the weather station WS1."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/attrs'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+
+  /types/{entityType}/attrs/{attrName}:
+    get:
+      operationId: reporter.reporter.query_1TNE1A
+      summary: "History of an attribute of N entities of the same type."
+      description: "For example, query the average pressure of this month in all the weather stations."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityTypeInPath'
+        - $ref: '#/parameters/attrNameInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/id'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
```

Comment:
what if two entities have the same index? should we keep track of the entities?

in case the reply could look like (and in case for entity on a given index there is not data, we use null):
```
{
     "entityType": "Room",
     "attrName": "temperature",
     "index": [1510826981388, 1510826990438, 1510826999725],
     "entities": [
          {
             "entityId": "Kitchen1",
             "values": [24.1, 25.3, 26.7]
          },
          {
            "entityId": "Room2",
             "values": [23.01, 24.9, 23.02]
          },
     ]
}
```
or it may be better to use separate indexes:

in case the reply could look like (and in case for entity on a given index there is not data, we use null):
```
{
     "entityType": "Room",
     "attrName": "temperature",
     "entities": [
          {
             "entityId": "Kitchen1",
             "index": [1510826981388, 1510826990438, 1510826999725],
             "values": [24.1, 25.3, 26.7]
          },
          {
            "entityId": "Room2",
            "index": [1510826981388, 1510826990438, 1510826999725],
             "values": [23.01, 24.9, 23.02]
          },
     ]
}
```

we should be able to specify if the aggregate is global or per entity.